internal/pikamon/config.go

```diff
@@ -0,0 +1,47 @@
+package pikamon
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Config contains the various values that are configurable for the bot.
+type Config struct {
+	Discord struct {
+		Token string
+	}
+	Logging struct {
+		Level string
+	}
+}
+
+// GetConfig reads the config file and flags, then applies environment variable overrides.
+func GetConfig() (*Config, error) {
+	cfg := &Config{}
+
+	// initialize config variables
+	viper.SetEnvPrefix("pikamon")
+	viper.SetConfigName("pikamon")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath("/etc/pikamon")
+	viper.AddConfigPath("./configs")
+
+	// read config file
+	err := viper.ReadInConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read config file")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "Failed to read config file " + viper.GetString(<>))
```
not sure what to put in `<>`...but basic idea is it would be nice to see the file it is trying to read when it fails to read it

Reply:
the error that is being wrapped contains the relevant information.